### PR TITLE
r-biobase: new version

### PR DESCRIPTION
**Pull request by baberlevi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-biobase/package.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ class RBiobase(RPackage):
 
     version('2.38.0', commit='83f89829e0278ac014b0bc6664e621ac147ba424')
     version('2.36.2', commit='15f50912f3fa08ccb15c33b7baebe6b8a59ce075')
+    version('2.40.0', commit='6555edbbcb8a04185ef402bfdea7ed8ac72513a5')
 
     depends_on('r-biocgenerics@0.16.1:', type=('build', 'run'))
-    depends_on('r@3.4.0:3.4.9', when='@2.36.2:')
+    depends_on('r@3.4.0:3.4.9', when='@2.36.2', type=('build', 'run'))
```
Comment (healther) [inline var/spack/repos/builtin/packages/r-biobase/package.py]: are these requirements hard? Typically we add the ranges, even though we might not have more versions available right now, they might be added later

Comment (baberlevi) [reply var/spack/repos/builtin/packages/r-biobase/package.py]: With bioconductor packages, there is an overall release of bioconductor (3.7 is current) which depends on a specific version of R. So the 2.36.2 version of this package which was part of bioconductor 3.5 depended on R 3.4, and the new version (part of bioconductor release 3.7) depends on R 3.5.

If we leave the unbounded upper range in place, I think the newer versions will still try to depend on R 3.4 ?

Comment (healther) [reply var/spack/repos/builtin/packages/r-biobase/package.py]: Yes an unbounded upper range will default to the newest available version. My question was more what happens if a `2.39.0` or a `2.38.1` release is added. Do you know on which version of `r` they will depend? From what you wrote I would expect it to be something like
```
depends_on('r@3.4.0:3.4.9', when='@2.36.0:2.36.99', type=('build', 'run'))
depends_on('r@3.5.0:3.5.9', when='@2.40.0:2.40.99', type=('build', 'run'))
```

Comment (baberlevi) [reply var/spack/repos/builtin/packages/r-biobase/package.py]: I don't know. The package versions without bioconductor seem arbitrary (not sure if sub versions cross R release versions )

**Pull request merged**